golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -78,22 +84,22 @@ func NewKeeper(
 // intermediate transaction state.
 //
 // The actionQueue's format is documented by `makeChainQueue` in
-// `packages/cosmic-swingset/src/chain-main.js`.
+// `packages/cosmic-swingset/src/make-queue.js`.
 func (k Keeper) PushAction(ctx sdk.Context, action vm.Jsonable) error {
 	bz, err := json.Marshal(action)
 	if err != nil {
 		return err
 	}
 
 	// Get the current queue tail, defaulting to zero if its vstorage doesn't exist.
-	tail := uint64(0)
-	tailStr := k.vstorageKeeper.GetData(ctx, "actionQueue.tail")
-	if len(tailStr) > 0 {
-		// Found, so parse it.
-		tail, err = strconv.ParseUint(tailStr, 10, 64)
-		if err != nil {
-			return err
-		}
+	tail, err := k.actionQueueIndex(ctx, "tail")
+	if err != nil {
+		return err
+	}
+
+	// JS uses IEEE 754 floats so avoid overflowing integers
+	if tail == MaxUint53 {
```

Comment:
Could be more defensive.

```suggestion
	if tail >= MaxUint53 {
```

Reply:
Arguably we're already busted then ;)